python/adbc_driver_snowflake/adbc_driver_snowflake/__init__.py

```diff
@@ -131,6 +131,7 @@ def connect(
     kwargs = (db_kwargs or {}).copy()
     if uri is not None:
         kwargs["uri"] = uri
+    kwargs[DatabaseOptions.APPLICATION_NAME] = "[ADBC][Python]" + kwargs.get(DatabaseOptions.APPLICATION_NAME, "")
```

Comment:
Use `dict.setdefault` so that we don't override the user choice?

Reply:
I wanted to prefix it to the user choice if they had one set